accepted/0022-git-buildpack.md

```diff
@@ -0,0 +1,64 @@
+# Git Support
+
+## Proposal
+
+Introduce an (optional) Git buildpack into each language buildpack family. The buildpack
+will be charged with providing `git` metadata as build-time and run-time environment
+variables.
+
+## Motivation
+
+Applications of any language may have a need for information extracted from
+`git` metadata. For example, the commit sha might be used as versioning
+information when building a Go binary or used when running a Rails application.
+
+## Implementation
+
+A new `git` buildpack will be developed to detect whether the `.git` directory and
+`BP_GIT_LOAD` environment variable is present and set to `true`, and then
+read the directory to extract the following environment variables so that they
+can be included in the built image.
+
+- `GIT_COMMIT_SHA`
+
+### Detection Criteria
+
+The buildpack will detect if:
+1. The `.git` directory is present
+2. The `BP_GIT_LOAD` environment variable is set to `true`.
```

Comment:
What is the purpose of this flag? Can we get away without it or would the buildpack do something unexpected or detrimental to the experience?